uktrade/data-hub-frontend
# Resource.Inline component

## Description of change

Adds the `Resource/Inline` component, which is also automatically attached to any of the resource components generated with the resource component factories so you can use it like this:

```js
import OrderInvoice from 'src/client/components/Resource/OrderInvoice'
import Event from 'src/client/components/Resource/Event'

<OrderInvoice.Inline id="opportunity-invoice-id">
  {opportunityInvoice => <pre>{JSON.stringify(opportunityInvoice)}</pre>}
</OrderInvoice.Inline>

<Event.Inline id="event-id">
  {event => <pre>{JSON.stringify(event)}</pre>}
</Event.Inline>
```

## Test instructions

1. Start storybook
2. Go to _Resource > Inline_ or `/?path=/story/resource--inline`
3. You should see a demo of appearance of the new `Task/Error.Inline` component in various contexts

## Screenshots

https://github.com/uktrade/data-hub-frontend/assets/2333157/6f9a5d4e-23db-4ac9-8364-b638615776e5

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
